src/solver/optimisation/constraints/ConstraintGroup.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "sim_structure_probleme_economique.h"
+#include "../opt_structure_probleme_a_resoudre.h"
+#include "constraint_builder_utils.h"
+
+class ConstraintGroup
+{
+public:
+    explicit ConstraintGroup(PROBLEME_HEBDO* problemeHebdo, ConstraintBuilder& builder) :
+     problemeHebdo_(problemeHebdo), builder_(builder)
+    {
+    }
+
+    virtual void Build() = 0;
+    PROBLEME_HEBDO* problemeHebdo_;
```

Comment:
`PROBLEME_HEBDO* problemeHebdo_;`
Eventually we also should remove it from class **ConstraintGroup**, but it's trickier than for **builder_**, because it's used in the constraint ordering loops (see **Group1::Build()** for example). We should try to find a way to extract these loops into classes.
So we'll leave it for now.